### PR TITLE
CVE-2014-3156 and CVE-2013-6655

### PR DESCRIPTION
**Pull request by emagliozzi**
CVEs for history project SWEN-331

### CODE REVIEW

File: cves/CVE-2013-6655.yml [hunk 1]
```diff
@@ -1,195 +1,205 @@
...
+  description later to get more technical.
+
+  Try to still be specific in your description, but remove Chromium-specific
+  stuff. Remove references to versions, specific filenames, and other jargon
+  that outsiders to Chromium would not understand. Technology like "regular
+  expressions" is fine, and security phrases like "invalid write" are fine to
+  keep too.
+description: This vulnerability is when a heap that had previously been freed,
+  would attempt to get accessed by scripts that were running as pages were being
+  loaded.
```
Comment (lbokun) [inline cves/CVE-2013-6655.yml]: What would be the result of someone taking advantage of this vulnerability? Also, use active voice.

File: cves/CVE-2013-6655.yml [hunk 2]
```diff
@@ -1,195 +1,205 @@
...
+    Was the original code unit tested, or not unit tested? Did the fix involve
+    improving the automated tests?
+
+    For the "code" answer below, look not only at the fix but the surrounding
+    code near the fix and determine if and was there were unit tests involved
+    for this module.
+
+    For the "fix" answer below, check if the fix for the vulnerability involves
+    adding or improving an automated test to ensure this doesn't happen again.
+  answer: yes
```
Comment (lbokun) [inline cves/CVE-2013-6655.yml]: Try to elaborate here. The fix commit did not have any obvious unit test files -- are they named something else? Were they in a nearby commit? The code and fix answers should be true or false.

File: cves/CVE-2013-6655.yml [hunk 3]
```diff
@@ -1,195 +1,205 @@
...
+  :note: Devs were trying to make loading start asynchronous... something to do
+    with beforeload events.
+upvotes_instructions: |
+  For the first round, ignore this upvotes number.
+
+  For the second round of reviewing, you will be giving a certain amount of
+  upvotes to each vulnerability you see. Your peers will tell you how
+  interesting they think this vulnerability is, and you'll add that to the
+  upvotes score on your branch.
+upvotes:
```
Comment (lbokun) [inline cves/CVE-2013-6655.yml]: +2

File: cves/CVE-2014-3156.yml [hunk 4]
```diff
@@ -1,197 +1,205 @@
...
+  description later to get more technical.
+
+  Try to still be specific in your description, but remove Chromium-specific
+  stuff. Remove references to versions, specific filenames, and other jargon
+  that outsiders to Chromium would not understand. Technology like "regular
+  expressions" is fine, and security phrases like "invalid write" are fine to
+  keep too.
+description: Inside of the clipboard implementation there was a buffer overflow
+  vulnerability. When trying to copy certain things to clipboard, the renderer
+  cliboard would return something with NULL pixels causing this overflow.
```
Comment (lbokun) [inline cves/CVE-2014-3156.yml]: What would be the result of someone taking advantage of this vulnerability?

File: cves/CVE-2013-6655.yml [hunk 5]
```diff
@@ -1,195 +1,205 @@
...
+    led to this vulnerability? Coding mistakes? Design mistakes?
+    Maintainability? Requirements? Miscommunications?
+
+    Look at the CWE entry for this vulnerability and examine the mitigations
+    they have written there. Are they doing those? Does the fix look proper?
+
+    Use those questions to inspire your answer. Don't feel obligated to answer
+    every one. Write a thoughtful entry here that those ing the software
+    engineering industry would find interesting.
+  answer: With my limited knowlege in the area, it sounds like this was really
```
Comment (lbokun) [inline cves/CVE-2013-6655.yml]: I agree with your conclusion

File: cves/CVE-2014-3156.yml [hunk 6]
```diff
@@ -1,197 +1,205 @@
...
+- 369621
+repo:
+fixes_vcc_instructions: |
+  Please put the commit hash in "commit" below (see my example in
+  CVE-2011-3092.yml). Fixes and VCCs follow the same format.
+fixes:
+- :commit: e4be88bde4a9b1f0d9a56bfc68447e0105657457
+  :note: ''
+vccs:
+- :commit: https://src.chromium.org/viewvc/chrome?revision=111097&view=revision
```
Comment (lbokun) [inline cves/CVE-2014-3156.yml]: Were you able to find the commit hash/id?

File: cves/CVE-2014-3156.yml [hunk 7]
```diff
@@ -1,197 +1,205 @@
...
+    Was the original code unit tested, or not unit tested? Did the fix involve
+    improving the automated tests?
+
+    For the "code" answer below, look not only at the fix but the surrounding
+    code near the fix and determine if and was there were unit tests involved
+    for this module.
+
+    For the "fix" answer below, check if the fix for the vulnerability involves
+    adding or improving an automated test to ensure this doesn't happen again.
+  answer: yes
```
Comment (lbokun) [inline cves/CVE-2014-3156.yml]: Try to elaborate here. The fix commit did not have any obvious unit test files -- are you referring to the files in the test shell directory? Were they in a nearby commit? The code and fix answers should be true or false.

File: cves/CVE-2014-3156.yml [hunk 8]
```diff
@@ -1,197 +1,205 @@
...
+- :commit: https://src.chromium.org/viewvc/chrome?revision=111097&view=revision
+  :note: Move clipboard-related webkit_glue embedder functions into a ClipboardClient interface.
+upvotes_instructions: |
+  For the first round, ignore this upvotes number.
+
+  For the second round of reviewing, you will be giving a certain amount of
+  upvotes to each vulnerability you see. Your peers will tell you how
+  interesting they think this vulnerability is, and you'll add that to the
+  upvotes score on your branch.
+upvotes:
```
Comment (lbokun) [inline cves/CVE-2014-3156.yml]: +3

File: cves/CVE-2014-3156.yml (hunk 6 above)
Comment (nicholastmosher) [reply cves/CVE-2014-3156.yml]: If that link points to the correct VCC, then i believe that the corresponding commit is [this one](https://github.com/chromium/chromium/commit/0de5d8609abf9961228c2a12031fc5354688345a). This is the first commit in which `content/renderer/renderer_clipboard_client.cc` appears, and we can see in the chromium review link that that file was created here.

File: cves/CVE-2014-3156.yml [hunk 9]
```diff
@@ -1,197 +1,205 @@
...
+  environment_variables:
+    applies:
+    note:
+  secure_by_default:
+    applies:
+    note:
+  yagni:
+    applies:
+    note:
+  complex_inputs:
```
Comment (nicholastmosher) [inline cves/CVE-2014-3156.yml]: This one may also be applicable. When dealing with complex inputs it's always safer to perform some assertions (e.g. that data is non-NULL) before working with them.

File: cves/CVE-2014-3156.yml (hunk 8 above)
Comment (nicholastmosher) [reply cves/CVE-2014-3156.yml]: :+1: :+1: :+1:
(+3)

**Pull request merged**